src/query/sql/src/planner/binder/window.rs

```
@@ -0,0 +1,379 @@
+                    column: ref ident,
+                    ..
+                } => {
+                    // We first search the identifier in select list
+                    let mut found = false;
+                    let database = database_name
+                        .as_ref()
+                        .map(|ident| normalize_identifier(ident, &self.name_resolution_ctx).name);
+                    let table = table_name
+                        .as_ref()
+                        .map(|ident| normalize_identifier(ident, &self.name_resolution_ctx).name);
+                    let column = normalize_identifier(ident, &self.name_resolution_ctx).name;
+
+                    for item in projections.iter() {
+                        if BindContext::match_column_binding(
+                            database.as_deref(),
+                            table.as_deref(),
+                            column.as_str(),
+                            item,
+                        ) {
+                            order_items.push(OrderItem {
+                                expr: order.clone(),
+                                index: item.index,
+                                name: item.column_name.clone(),
+                                need_eval_scalar: scalar_items.get(&item.index).map_or(
+                                    false,
+                                    |scalar_item| {
+                                        !matches!(
+                                            &scalar_item.scalar,
+                                            ScalarExpr::BoundColumnRef(_)
+                                        )
+                                    },
+                                ),
+                            });
+                            found = true;
+                            break;
+                        }
+                    }
+
+                    if found {
+                        continue;
+                    }

```

I think we don't need to do this for `order by` in window functions. It's different from `order by` clause in the query body, we can just throw an error if no matched column is found.